Packs/FeedInfobloxThreatIntelligence/Integrations/FeedInfobloxThreatIntelligence/FeedInfobloxThreatIntelligence.py

```diff
@@ -0,0 +1,946 @@
+            raise DemistoException("Authentication Error (401): API key is invalid or expired.")
+        elif status_code == 403:
+            raise DemistoException("Forbidden (403): Insufficient permissions to access this resource.")
+        elif status_code == 404:
+            raise DemistoException("Not Found (404): The requested resource was not found.")
+        elif status_code == 429:
+            raise DemistoException("Rate Limit Exceeded (429): Too many requests. Please try again later.")
+        elif 500 <= status_code < 600:
+            raise DemistoException(f"Server Error ({status_code}): Internal server error occurred. Please try again later.")
+        else:
+            err_msg = HTTP_ERRORS.get(status_code, f"Error in API call with status code {status_code}.")
+            raise DemistoException(err_msg)
+
+    def http_request(
+        self,
+        method: str,
+        url_suffix: str = "",
+        full_url: str = None,
+        params: dict = None,
+        data: dict = None,
+        json_data: dict = None,  # noqa: ARG002
+        headers: dict = None,
+        timeout: int = None,
+    ) -> Any:
+        """Send an HTTP request with enhanced error handling and retry mechanism.
+
+        Args:
+            method (str): HTTP method (GET, POST, etc.).
+            url_suffix (str, optional): Suffix to append to the base URL. Defaults to ''.
+            full_url (str, optional): Full URL to use instead of base URL + suffix. Defaults to None.
+            params (dict, optional): URL parameters. Defaults to None.
+            data (dict, optional): Form data. Defaults to None.
+            json_data (dict, optional): JSON data body. Defaults to None.
+            headers (dict, optional): Additional headers. Defaults to None.
+            timeout (int, optional): Request timeout. Defaults to None.
+
+        Returns:
+            Any: Response from the API.
+
+        Raises:
+            DemistoException: If the request fails.
+        """
+        # Log the request details for debugging
+        demisto.debug(f"Requesting Infoblox TIDE API with method: {method}, url_suffix: {url_suffix} and params: {params}")
+
+        try:
+            res = self._http_request(
+                method=method,
+                url_suffix=url_suffix,
+                full_url=full_url,
+                params=params,
+                data=json.dumps(data) if data else None,
+                headers=headers,
+                timeout=timeout or self._read_timeout,
+                resp_type="response",
+                ok_codes=(200, 201, 204),
+                error_handler=self._handle_error_response,
+                retries=TOTAL_RETRIES,
+                status_list_to_retry=STATUS_CODE_TO_RETRY,
+                backoff_factor=BACKOFF_FACTOR,
+            )
+
+            status_code = res.status_code
+
+            # Handle specific status codes with detailed error messages
+            if status_code not in VALID_CODES:
```

Comment:
This block of error handling logic is redundant and inconsistent with the `_handle_error_response` method. It's recommended to centralize error handling in one place.